src/ledger/NetworkConfig.h

```diff
@@ -18,13 +18,18 @@ namespace stellar
 struct InitialSorobanNetworkConfig
 {
     // Contract size settings
-    static constexpr uint32_t MAX_CONTRACT_SIZE = 65536;
+    static constexpr uint32_t MAX_CONTRACT_SIZE = 64 * 1024; // 64KB
+
+    // Contract data settings
+    static constexpr uint32_t MAX_CONTRACT_DATA_KEY_SIZE_BYTES = 300;
+    static constexpr uint32_t MAX_CONTRACT_DATA_ENTRY_SIZE_BYTES =
+        64 * 1024; // 64KB
 
     // Compute settings
     static constexpr int64_t LEDGER_MAX_INSTRUCTIONS = 1;
     static constexpr int64_t TX_MAX_INSTRUCTIONS = 200'000'000;
     static constexpr int64_t FEE_RATE_PER_INSTRUCTIONS_INCREMENT = 1;
-    static constexpr uint32_t MEMORY_LIMIT = 50 * 1024 * 1024; // 50MB
+    static constexpr uint32_t MEMORY_LIMIT = 250 * 1024 * 1024; // 250MB
```

Comment:
This is a change corresponding to the the TX_MAX_INSTRUCTIONS change from 40M->200M.
Since instructions and memory are mostly tied together (vm instantiation costs 1M for each), if we don't bump this one  someone will reach the memory limit way before getting close to the 200M instruction limit.

Reply:
This number does appear to be large. But it's not peak memory but rather cumulative. 
If we are letting people submit multiple independent host invocations sharing the same budget, then both instruction and memory limits needs to be raised.

Reply:
> Since instructions and memory are mostly tied together (vm instantiation costs 1M for each),

I don't necessarily agree here. Instruction limit is dictated by the target ledger close time while memory limit is dictated by the core instance RAM capacity. In theory the instruction limit could be equal to per-ledger instruction limit, that just would have issues with spam potentially taking over ledgers and also that wouldn't parallelize well. As for memory, we should consider multi-threaded execution. E.g. executing 8 transactions in parallel would require 2GB RAM at peak etc. I'm not sure if that's much, but I'd rather be a bit more conservative here, like maybe 100MB for now?

> But it's not peak memory but rather cumulative.

I'm not sure if that's true as we don't do any garbage collection, so all the host objects will stay alive for the whole execution time.

> If we are letting people submit multiple independent host invocations sharing the same budget, then both instruction and memory limits needs to be raised.

I just wanted to reiterate that allowing multiple host fn invocations is purely a convenience feature - it could be emulated via a simple contract that batches ops together. There is no intention of proportionally increasing the resource limits.

Reply:
I get your point about the ledger capacity vs core RAM. 

> But it's not peak memory but rather cumulative.
> > I'm not sure if that's true as we don't do any garbage collection
The VM instance is freed up after each invocation, so the mem cost of independent invocations only happen one at a time (not considering parallel execution). 

> There is no intention of proportionally increasing the resource limits.

I'm wondering if we should just to keep the resource limit at 40M instructions and 50MB memory (both same as before) then? I agree we should start conservative and this is what we had before and has worked.

Reply:
> The VM instance is freed up after each invocation, so the mem cost of independent invocations only happen one at a time (not considering parallel execution).

Interesting, but the host objects will stay in memory, right?

> I'm wondering if we should just to keep the resource limit at 40M instructions and 50MB memory (both same as before) then? I agree we should start conservative and this is what we had before and has worked.

Yeah, that's an option as well. We can see if the limit is hit frequently and we'll also need to do more tuning anyway.

Reply:
The frame will be popped at the end of the invocation so the host objects will be freed as well. 

I have changed the limits back to 40M cpu and 50M mem, we can go back to tuning these values as you said.